truffle/src/main/ruby/core/encoding.rb

```diff
@@ -141,7 +141,7 @@ class Converter
     def self.asciicompat_encoding(string_or_encoding)
       encoding = Rubinius::Type.try_convert_to_encoding string_or_encoding
 
-      return if not encoding or undefined.equal? encoding
+      return unless encoding
```

Comment:
Doesn't `BooleanCastNode` need to be updated to handle the `undefined` object in order for this to work?